blog/2023-07-18-wing-cloud-launch.md

```diff
@@ -0,0 +1,281 @@
+Ventures](https://www.grovevc.com/) and [StageOne Ventures](https://stageonevc.com/) as well as an
+incredible crew of [funds and angels](TBD) from across the industry on the journey to turn this idea
+into a reality.
+
+**So what is Wing Cloud**? It is a new kind of *abstract cloud*. It doesn’t involve data centers,
+machines, or provisioning engines. Instead, it’s a layer that enable companies to utilize the cloud
+as a general-purpose computing platform. It does that through a programming and operational model
+that spans both infrastructure and application across all cloud providers and services.
+
+Wing Cloud is a unified programming and operational model that spans all providers and services. It
+celebrates the choice and flexibility the cloud has to offer without compromising the developer
+experience and engineering practices.
+
+So how does that look like in practice? Our vision for Wing Cloud consists of three main components:
+
+1. A programming language for the cloud, aptly called **Winglang**.
+2. An high level cloud SDK, aptly called **Wing Cloud Library**.
+3. An application-centric operational console, aptly called **Wing Console**.
+
+### Why are we building a programming language?
+
+With **Winglang**, developers are able to write code that includes both infrastructure **and**
+runtime code. This means that when a cloud resource is needed, developers can just add it to their
+ code and interact with it in the same way they interact with in-memory objects in traditional
+languages. The Winglang compiler takes care of the mechanics required to provision the resource,
+configure the minimal IAM permissions and wire up the infrastructure configuration so your code can
+interact with this resource at runtime.
+
+You might be wondering why this couldn't be implemented as a library within an existing language.
+This is because existing languages don't have language constructs that can express the distributed
+nature of cloud applications. To address this, Winglang has two execution phases: *preflight* and
+*inflight*. The preflight phase defines your application's infrastructure and the inflight phase
+defines its runtime behavior.
+
+Another way to think about it is *space* versus *time*. Think about programming languages as
+storytelling devices. Traditional languages are designed to tell a linear story: first this
+happened, then that, and so on. Basically the traditional purpose of programming languages was to
+describe a sequence of instructions for the CPU. They tells the story of a program over **time**.
+
+But when it comes to cloud applications, the story is not linear. There are multiple storylines
+running together, interacting with and affecting each other. It's not just about what happens
+*when*, but also *where*. There's a *spatial dimension* to account for, and it's a dimension our
+current languages are just don't designed to express.
```

Comment:
This space/time analogy didn't really click to me on first read. Even after thinking about it for a little longer, I feel a little confused since programs from traditional languages also deal with space and time requirements, right?

Are you trying to express that preflight and inflight correspond to space and time, or that preflight and inflight are orthogonal dimensions/concerns in the same way that space and time are? I wasn't sure totally sure

Reply:
I agree, I think it will not click with many readers because they will compare this to async programing for single machines that also has multiple stories at the same time. 
The difference to the cloud is in the spacial dimension, but it is not really explained here, it is just mentioned without really explaining how it differs from regular async programming for single machines and why it matters on the cloud and needs a new language to support it.

Reply:
OK, I removed the analogy but I am a bit sad to let it go because it was a good one. LMK what you think.

Reply:
Yeah, I didn't mean to prescribe a solution (whether to remove it or tweak it etc). Maybe there's some way to rephrase it?